src/fstext/kaldi-fst-io.cc

```diff
@@ -42,6 +42,12 @@ VectorFst<StdArc> *ReadFstKaldi(std::string rxfilename) {
   return fst;
 }
 
+// Register const fst to load it automatically. Other types like
+// olabel_lookahead or ngram or compact_fst should be registered
+// through OpenFst registration API.
+static fst::FstRegisterer<VectorFst<StdArc>> VectorFst_StdArc_registerer;
```

Comment:
Can you explain what this is for and how it worked previously?

Reply:
FST has a registration mechanism (http://www.openfst.org/doxygen/fst/html/register_8h.html) for loading the FST depending on the header type. If type is registered, you can simply load FST - vector, const, lookahead, ngram as

`Fst<StdArc> *fst = Fst<StdArc>::Read(ki.Stream(), ropts);
`
And when you link to extra so files, types are updated automatically.

Before that it was explicit

  ```
FstReadOptions ropts("<unspecified>", &hdr);
  Fst<StdArc> *fst = NULL;
  if (hdr.FstType() == "const") {
    fst = ConstFst<StdArc>::Read(ki.Stream(), ropts);
  } else if (hdr.FstType() == "vector") {
    fst = VectorFst<StdArc>::Read(ki.Stream(), ropts);
  }
```

and we will have to add more cases like:

  else if (hdr.FstType() == "ngram") {
    fst = NgramFst<StdArc>::Read(ki.Stream(), ropts);
  }

etc.

Reply:
how to register  olabel_lookahead type

Fst::Read: Unknown FST type olabel_lookahead (arc type = standard)